pkg/operation/botanist/component/seedadmissioncontroller/seedadmissioncontroller.go

```diff
@@ -174,16 +181,25 @@ func (g *gardenerSeedAdmissionController) Deploy(ctx context.Context) error {
 			},
 		}
 
-		deployment = &appsv1.Deployment{
+		// if maxUnavailable would not be set, new pods don't come up in small seed clusters
+		// (due to the pod anti affinity new pods are stuck in pending state)
+		maxUnavailable = intstr.FromInt(1)
```

Comment:
Could this be `0` if the number of nodes is grater than the replicas?

Reply:
I think so, yes, but it also enforces a re-distribution of the pods onto other nodes. I guess, when rolling the pods one by one the `maxUnvailable: 1` configuration should not be too big of a problem regarding time of deployment. What do you think? Do you want default settings back for seed clusters > 3 nodes?